packages/remix-dev/compiler/assets.ts

```diff
@@ -85,22 +88,27 @@ export async function createAssetsManifest(
         /(^browser-route-module:|\?browser$)/g,
         ""
       );
-      let route = routesByFile.get(entryPointFile);
-      invariant(route, `Cannot get route for entry point ${output.entryPoint}`);
-      let sourceExports = await getRouteModuleExportsCached(config, route.id);
-      routes[route.id] = {
-        id: route.id,
-        parentId: route.parentId,
-        path: route.path,
-        index: route.index,
-        caseSensitive: route.caseSensitive,
-        module: resolveUrl(key),
-        imports: resolveImports(output.imports),
-        hasAction: sourceExports.includes("action"),
-        hasLoader: sourceExports.includes("loader"),
-        hasCatchBoundary: sourceExports.includes("CatchBoundary"),
-        hasErrorBoundary: sourceExports.includes("ErrorBoundary"),
-      };
+      let groupedRoute = routesByFile.get(entryPointFile);
+      invariant(
+        groupedRoute,
+        `Cannot get route(s) for entry point ${output.entryPoint}`
+      );
+      for (let route of groupedRoute) {
+        let sourceExports = await getRouteModuleExportsCached(config, route.id);
+        routes[route.id] = {
```

Comment:
> what is the best way to "verbatim and error if they provide duplicates" just throws a new Error inside routes.ts file?

And then add an invariant here to check for duplicates?

```js
invariant(routes[route.id] == null, `Found duplicate route definitions for id "${route.id}"`);
```

Reply:
There are some possibility that we don't need to check for duplicity? Maybe sometimes user could just overwrite some route by using same ID booth situations...

Maybe we don't need to check at all just remove the `{filename}--{id}` thing and will be fine?

Reply:
We'll need to check so we get a quick built time error instead of a runtime error inside of the router. ID's must be unique.

Reply:
So @jacob-ebey the best place to check this duplicates is in `packages/remix-dev/config/routes.ts` right?